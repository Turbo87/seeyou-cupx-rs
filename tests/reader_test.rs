use insta::assert_compact_debug_snapshot;
use seeyou_cupx::CupxFile;
use std::io::Read;

#[test]
fn test_westalpen() {
    let (mut cupx, warnings) = CupxFile::from_path("tests/fixtures/westalpen_de.cupx").unwrap();
    assert_eq!(cupx.waypoints().len(), 126);
    assert_eq!(cupx.tasks().len(), 0);
    assert_eq!(warnings.len(), 0);

    let mut image_files = cupx.picture_names().collect::<Vec<_>>();
    image_files.sort();
    insta::assert_debug_snapshot!(image_files);

    let mut reader = cupx.read_picture("2_1034.jpg").unwrap();
    let mut buffer = Vec::new();
    reader.read_to_end(&mut buffer).unwrap();
    insta::assert_binary_snapshot!("2_1034.jpg", buffer);
}

#[test]
fn test_ec25_no_pictures_zip() {
    let (cupx, warnings) = CupxFile::from_path("tests/fixtures/EC25_no_pictures_zip.cupx").unwrap();
    assert_eq!(cupx.waypoints().len(), 221);
    assert_eq!(cupx.tasks().len(), 0);
    assert_compact_debug_snapshot!(warnings, @"[NoPicturesArchive]");
    assert_eq!(cupx.picture_names().count(), 0);
}
