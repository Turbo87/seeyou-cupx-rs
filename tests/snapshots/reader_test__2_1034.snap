---
source: tests/reader_test.rs
expression: buffer
extension: jpg
snapshot_kind: binary
---
